bin/compile

```diff
@@ -16,7 +16,11 @@ arrow "Setting up .profile.d to automatically run agentmon..."
 mkdir -p "${BUILD_DIR}/.profile.d"
 cp .profile.d/heroku-metrics-daemon.sh "${BUILD_DIR}/.profile.d"
 
-if [[ -f "${BUILD_DIR}/pom.xml" ]] || [[ -f "${BUILD_DIR}/build.gradle" ]] || [[ -f "${BUILD_DIR}/build.sbt" ]] || [[ -d "${BUILD_DIR}/.jdk" ]]; then
+if [[ -f "${BUILD_DIR}/pom.xml" ]] ||
```

Comment:
There is no "return code" as such in Bash. only exit codes. So you can't return a string, which is why you see functions echo a string and then the caller captures it with `s="$(my_func)"`. bash is cra cra. I'll submit another PR with some refactoring changes like this.